lib/core/include/irods/irods_load_plugin.hpp

```diff
@@ -94,6 +101,274 @@ namespace irods
+                    }
+                }
+                else {
+                    Elf64_Shdr s_hdr_in{};
+                    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+                    lib.read(reinterpret_cast<char*>(&s_hdr_in), sizeof(s_hdr_in));
+                    if (e_shnum == 0) {
+                        e_shnum = s_hdr_in.sh_size;
+                    }
+                    if (e_shstrndx == SHN_XINDEX) {
+                        e_shstrndx = s_hdr_in.sh_link;
+                    }
+                }
+            }
+
+            // offset of string table section header
+            const std::uint64_t e_shstroff = e_shoff + (static_cast<std::uint64_t>(e_shstrndx) * e_shentsize);
+            // seek to string table section header
+            lib.seekg(e_shstroff);
+
+            std::uint64_t e_strtoff{};
+            std::uint64_t e_strtsz{};
+            if (ei_class == ELFCLASS32) {
+                Elf32_Shdr s_hdr_in{};
+                // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+                lib.read(reinterpret_cast<char*>(&s_hdr_in), sizeof(s_hdr_in));
+                e_strtoff = s_hdr_in.sh_offset;
+                e_strtsz = s_hdr_in.sh_size;
+            }
+            else {
+                Elf64_Shdr s_hdr_in{};
+                // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+                lib.read(reinterpret_cast<char*>(&s_hdr_in), sizeof(s_hdr_in));
+                e_strtoff = s_hdr_in.sh_offset;
+                e_strtsz = s_hdr_in.sh_size;
+            }
+
+            // seek to section name string table
+            lib.seekg(e_strtoff);
+
+            std::vector<char> e_strtbl_vec(e_strtsz);
+            lib.read(e_strtbl_vec.data(), e_strtsz);
+
+            // TODO: some platforms do not include a section header for the dynamic table
+            // TODO: some platforms do not mark the dynamic table section header as SHT_DYNAMIC
+            for (std::uint64_t sh_idx = 1; sh_idx < e_shnum; sh_idx++) {
+                // seek to section header
+                lib.seekg(e_shoff + (sh_idx * e_shentsize));
+
+                std::uint32_t sh_name{};
+                std::uint32_t sh_type{};
+                std::uint64_t sh_offset{};
+                std::uint64_t sh_size{};
+                std::uint64_t sh_entsize{};
+                std::string_view s_name{};
+                if (ei_class == ELFCLASS32) {
+                    Elf32_Shdr s_hdr_in{};
+                    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+                    lib.read(reinterpret_cast<char*>(&s_hdr_in), sizeof(s_hdr_in));
+
+                    sh_type = s_hdr_in.sh_type;
+                    if (sh_type != SHT_DYNAMIC) {
+                        continue;
+                    }
+                    sh_name = s_hdr_in.sh_name;
+                    s_name = std::string_view(&e_strtbl_vec.data()[sh_name]);
```

Comment:
So `&e_strtbl_vec[sh_name]` doesn't work?

Is the pointer returned null-terminated?

Reply:
> So `&e_strtbl_vec[sh_name]` doesn't work?

My understanding is that depends on the implementation of `std::vector`, but I could be mistaken.

> Is the pointer returned null-terminated?

Yes.